uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/LeaveRequestComment.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+use CRM_HRLeaveAndAbsences_BAO_LeaveRequest as LeaveRequest;
+use CRM_HRLeaveAndAbsences_Service_LeaveManager as LeaveManagerService;
+
+class CRM_HRLeaveAndAbsences_Service_LeaveRequestComment {
+
+  /**
+   * @var string
+   */
+  private $entityName = 'LeaveRequest';
+
+  /**
+   * @var array
+   *   The parameter array needed to add, create or delete comments
+   */
+  private $params;
+
+  /**
+   * CRM_HRLeaveAndAbsences_Service_LeaveRequestComment constructor.
+   *
+   * @param $params
+   *   The parameter array needed to add, create or delete comments
+   */
+  public function __construct($params) {
```

Comment:
I don't think $params should be passed on the constructor. The reasons are:
- You have to instantiate the service for every single action you will want to executed. For example, if you want to create 10 comments, you'll need 10 service instances and 10 calls to the add method.
- It will make thinks break. The params required for the `delete` method, are different from those of the `add` or the `get` methods. What happens if I instantiate the service with only the `comment_id` but then I call the `add` method?

I would remove `$params` from `__construct` and make it a param on each of the methods